### ISSUE TITLE
[Source Request] Komga

### ISSUE THREAD

**Issue by h-wb** [Paperback-iOS/extensions#103]
I'd love to see Komga as an extension for Paperback. [Komga](https://komga.org/) is a free and open source comics/mangas media server. There's an [existing extension for Tachiyomi](https://github.com/inorichi/tachiyomi-extensions/tree/master/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga) which could be used as a model and a [public and documented REST Api](https://komga.org/guides/rest.html).

I opened an issue on the kogma repo [here](https://github.com/gotson/komga/issues/263#issue-674956703) but unfortunately the Komga dev does not own any iOS devices. I could see myself doing it but I'm not sure, in this state of the app is it possible for me to test an extension I'm creating ?

**Comment by FramboisePi** [comment]
Hello,
I also believe it would be great for Paperback to have a Komga extension, thank you for the suggestion. 

It is indeed possible to test an extension in the app while developing it, you just need Paperback beta, in order to be able to add external sources. You can also execute a test file on your computer. Just note that the documentation is… not really complete…

There may just be two issues with the current state of development of Paperback:
I don’t think sources are able to register a setting to let the user choose its own Komga server and credentials.

Unless you want to do the extension yourself, I would be really happy to try to work on it.

After reading Komga documentation, I have a question: how are mangas stored in Komga? 
If I understood correctly a `series` would be used to represent a manga and the series `books` to represent the manga chapters. That would mean a manga can not have a description or an author… These information would have to be found in the chapters (`books`) of the manga (`series`) in question.

**Comment by h-wb** [comment]
> Hello,
> I also believe it would be great for Paperback to have a Komga extension, thank you for the suggestion.
> 
> It is indeed possible to test an extension in the app while developing it, you just need Paperback beta, in order to be able to add external sources. You can also execute a test file on your computer. Just note that the documentation is… not really complete…
> 
> There may just be two issues with the current state of development of Paperback:
> I don’t think sources are able to register a setting to let the user choose its own Komga server and credentials.
> 
> Unless you want to do the extension yourself, I would be really happy to try to work on it.

Sure, go ahead ! I was willing to do it if no else would but I would probably take a lot of time aha..

> After reading Komga documentation, I have a question: how are mangas stored in Komga?
> If I understood correctly a `series` would be used to represent a manga and the series `books` to represent the manga chapters. That would mean a manga can not have a description or an author… These information would have to be found in the chapters (`books`) of the manga (`series`) in question.

I opened an issue on the Komga repo, I asked your question [here](https://github.com/gotson/komga/issues/263#issuecomment-673442807), Gotson will give you a better answer than me !

**Comment by gotson** [comment]
@FramboisePi you're correct in your understanding, and that's how the Tachiyomi extension works currently, a Series is a manga, and a Book is a chapter.

It is also correct that at the moment there is not description or authors per series. There are planned enhancements for that. At the moment in Tachiyomi, the information for those fields is left blank.

Feel free to ask me any question on Komga, either here, on the corresponding Komga issue, or via Discord.

**Comment by FramboisePi** [comment]
Thank you for the answer. I will try to understand Tachiyomi extension to see if I can use some ideas in the source.
I will definitely join the Discord server and I will see if it is possible to get information from the first chapter to fill the description.

---

I have a working source that is able to show new and updated series, search for manga by title ans status, and display a title content.

There are still two issues
* Thumbnails do not work, there may be an authorization issue that should be in investigation.
* It is not possible to choose a Komga server in the app and must be hard coded in the source. This will be possible in a later version of Paperback. 

If someone is interested by this partially working source, I can share it. I will finish it when needed feature will be supported by Paperback.

**Comment by gotson** [comment]
The thumbnails need the same Basic Auth than the API, that may be your problem.

You can setup your own Komga server locally for debugging, that way you can enable extra debug logs on the web layer to catch those kind of errors server side. 

Hopefully paperback will get extension configuration soon!

**Comment by ConradWeiser** [comment]
Thanks to @FramboisePi, Komga is (I believe) completed. The source will not work until Paperback version 0.6 is released, but this can be marked as confirmed, and pretty much ready.
